File: kinotic-os-api/src/main/java/org/kinotic/os/api/model/iam/PendingInviteSummary.java
package org.kinotic.os.api.model.iam;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

import java.util.Date;

/**
 * View of a pending invitation for listing in the web app. Deliberately excludes the accept token — anyone who
 * can list invitations must not be able to redeem them.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class PendingInviteSummary {

    private String id;

    /** Email the invitation was sent to. */
    private String email;

    /** Display name the inviter entered, if any. */
    private String displayName;

    /** The Application the invitee would join; {@code null} for an organization-member invite. */
    private String applicationId;

    /** Display name of the member who sent the invitation. */
    private String invitedByName;

    private Date created;

    /** When the invitation stops being acceptable. */
    private Date expiresAt;
}

File: kinotic-js/workspace/packages/os-api/src/api/model/iam/PendingInviteSummary.ts
import type { Identifiable } from '@kinotic-ai/core'

/**
 * Console view of a pending invitation. Deliberately excludes the accept token — anyone who
 * can list invitations must not be able to redeem them.
 */
export class PendingInviteSummary implements Identifiable<string> {
    public id: string | null = null

    /** Email the invitation was sent to. */
    public email: string = ''

    /** Display name the inviter entered, if any. */
    public displayName: string | null = null

    /** The application the invitee would join; null for an organization-member invite. */
    public applicationId: string | null = null

    /** Display name of the member who sent the invitation. */
    public invitedByName: string | null = null

    public created: number | null = null

    /** When the invitation stops being acceptable. */
    public expiresAt: number | null = null
}
